src/studio/ui/StudioPanel.java

```diff
@@ -741,8 +796,14 @@ else if ((row == 1) && ((noWins % 3) > 1))
         }
     }
 
-    private void setServer(Server server) {
-        if (server == null) return;
+    private void setServer(Server server, boolean addToNavigationHistory) {
+        if (server == null)
+            return;
+
+        if (addToNavigationHistory)
+            serverNavigateHistory.addSelectedItem(server);
+        prevServerAction.setEnabled(serverNavigateHistory.hasPreviousItem());
```

Comment:
I want to put logic for actions enable/disable into a single place - refreshActionState()